connector_search_engine/models/se_binding.py

```diff
@@ -64,6 +63,14 @@ def get_export_data(self):
         """Public method to retrieve export data."""
         return self.data
 
+    def _is_indexed(self):
+        self.ensure_one()
+        if self.sync_state == "new":
+            return False
+        elif self.sync_state == "done" and not self.active:
+            return False
```

Comment:
@ivantodorovich  Why keep the test on active=False ? 

I guess the goal of this PR is to always allow the unlink() and send a DELETE to SE only if needed.

Then, the only relevant test here is to check for sync_state.


PS: currently: active=False and state=done are unlinked from odoo but never deleted from the SE

Reply:
Hello @hparfr !

Because those records should've been deleted by the `record.exporter.deleter` in fact

When a record is archived, it's flagged for synchronization [^1]
When synchronized, the deleted removes it from the index [^2] [^3]

That should be enough but maybe I'm missing something?

[^1]: https://github.com/OCA/search-engine/blob/949c1f52143c5157fec75e94de5ecec823630817/connector_search_engine/models/se_binding.py#L82-L86
[^2]: https://github.com/OCA/search-engine/blob/949c1f52143c5157fec75e94de5ecec823630817/connector_search_engine/components/deleter.py#L7-L22
[^3]: https://github.com/OCA/search-engine/blob/949c1f52143c5157fec75e94de5ecec823630817/connector_search_engine/models/se_binding.py#L196-L199

Reply:
Ok, I miss the case when the record is not anymore in the index. (active=False and done=true). It's looks a bit fragile, may be it's better to have a new state "deleted" instead.

But back to this point. The actual implementation do not work when active is set in a compute method because write() is not called. Example : https://github.com/shopinvader/odoo-shopinvader/blob/14.0/shopinvader/models/shopinvader_variant.py#L76

I try a fix in #129 , it needs testing.